cmd/meroxa/turbine/utils.go

```diff
@@ -182,18 +184,43 @@ func GitInit(ctx context.Context, appPath string) error {
 		return errors.New("path is required")
 	}
 
-	cmd := exec.Command("git", "config", "--global", "init.defaultBranch", "main")
-	cmd.Path = appPath
-	_ = cmd.Run()
+	oldWay := checkGitVersion(ctx)
```

Comment:
I think `oldWay` is not explanatory enough to know what it does unless checking the `checkGitVersion` function. What do you think we change it to something like this?:

I also made some changes to ensure we return err if it's found when doing `cmd.Run`.

```go
func GitInit(ctx context.Context, appPath string) error {
	if appPath == "" {
		return errors.New("path is required")
	}
      
       isGitOlderThan228 := checkGitVersion(ctx)

	if !isGitOlderThan228 {
		cmd := exec.CommandContext(ctx, "git", "config", "--global", "init.defaultBranch", "main")
		cmd.Path = appPath
		err := cmd.Run()
                 if err != nil {
                      return err
                 }
	}

	cmd := exec.CommandContext(ctx, "git", "init", appPath)
	output, err := cmd.CombinedOutput()
	if err != nil {
		return fmt.Errorf(string(output))
	}

	if isGitOlderThan228 {
		cmd := exec.CommandContext(ctx, "git", "checkout", "-b", "main")
		cmd.Path = appPath
		_ = cmd.Run()
	}
	return nil
}
```

Reply:
looking at the workflow again... this is the last step in init.  if the customer was savvy enough, they could figure out what was left to do and do it themselves. it's not too annoying. i'll update.